geoutils/pointcloud/pointcloud.py

```diff
@@ -1 +1,393 @@
+
+    # Read file
+    las = laspy.read(filename)
+
+    # Get data from requested columns
+    data = np.vstack([las[n] for n in columns]).T
+
+    # Build geodataframe
+    gdf = gpd.GeoDataFrame(
+        geometry=gpd.points_from_xy(x=las.x, y=las.y, crs=las.header.parse_crs(prefer_wkt=False)),
+        data=data,
+        columns=columns,
+    )
+
+    return gdf
+
+
+def _load_laspy_metadata(
+    filename: str,
+) -> tuple[CRS, int, BoundingBox, pd.Index]:
+    """Load point cloud metadata from LAS/LAZ/COPC file."""
+
+    with laspy.open(filename) as f:
+
+        crs = f.header.parse_crs(prefer_wkt=False)
+        nb_points = f.header.point_count
+        bounds = BoundingBox(left=f.header.x_min, right=f.header.x_max, bottom=f.header.y_min, top=f.header.y_max)
+        columns_names = pd.Index(list(f.header.point_format.dimension_names))
+
+    return crs, nb_points, bounds, columns_names
+
+
+# def _write_laspy(filename: str, pc: gpd.GeoDataFrame):
+#     """Write a point cloud dataset as LAS/LAZ/COPC."""
+#
+#     with laspy.open(filename) as f:
+#         new_hdr = laspy.LasHeader(version="1.4", point_format=6)
+#         # You can set the scales and offsets to values that suits your data
+#         new_hdr.scales = np.array([1.0, 0.5, 0.1])
+#         new_las = laspy.LasData(header = new_hdr, points=)
+#
+#     return
+
+
+class PointCloud(gu.Vector):  # type: ignore[misc]
+    """
+    The georeferenced point cloud.
+
+    A point cloud is a vector of 2D point geometries associated to numeric values from a data column, and potentially
+    auxiliary data columns.
+
+     Main attributes:
+        ds: :class:`geopandas.GeoDataFrame`
+            Geodataframe of the point cloud.
+        data_column: str
+            Name of point cloud data column.
+        crs: :class:`pyproj.crs.CRS`
+            Coordinate reference system of the point cloud.
+        bounds: :class:`rio.coords.BoundingBox`
+            Coordinate bounds of the point cloud.
+
+
+    All other attributes are derivatives of those attributes, or read from the file on disk.
+    See the API for more details.
+    """
+
+    def __init__(
+        self,
+        filename_or_dataset: str | pathlib.Path | gpd.GeoDataFrame | gpd.GeoSeries | BaseGeometry,
+        data_column: str,
+    ):
+        """
+        Instantiate a point cloud from either a data column name and a vector (filename, GeoPandas dataframe or series,
+        or a Shapely geometry), or only with a point cloud file type.
+
+        :param filename_or_dataset: Path to vector file, or GeoPandas dataframe or series, or Shapely geometry.
+        :param data_column: Name of main data column defining the point cloud.
+        """
+
+        self._ds: gpd.GeoDataFrame | None = None
+        self._name: str | None = None
+        self._crs: CRS | None = None
+        self._bounds: BoundingBox
+        self._data_column: str
+        self._data: NDArrayNum
+        self._nb_points: int
```

Comment:
Intuitively I would have called this "count", which I think is often used by other tools. But this may be ambiguous with the `Raster.count` which is about band count.